lib/absinthe/execution/arguments.ex

```diff
@@ -106,15 +106,14 @@ defmodule Absinthe.Execution.Arguments do
   end
 
   defp retrieve_variable(name, schema_type, type_stack, ast, meta) do
-    full_type_stack = fillout_stack(schema_type, [], meta.schema)
-    meta.variables
-    |> Map.get(name)
-    |> case do
+    full_type_stack = fillout_stack(schema_type.type, [], meta.schema)
+    
+    case Map.get(meta.variables, name) do
       # The variable exists, and it has the same
       # type as the argument in the schema.
       # yay! we can use it as is.
       %{value: value, type_stack: ^full_type_stack} ->
-        do_retrieve_variable(value, schema_type, type_stack, ast, meta)
+        do_retrieve_variable(value, schema_type.type, type_stack, ast, meta)
```

Comment:
So one issue here is that a whole lot of types don't have a type field, so this isn't gonna work. schema_type isn't the variable, it's the type from the schema that the variable is supposed to be.